community/assessments/projects/kubescape/self-assessment.md

```diff
@@ -0,0 +1,323 @@
+* Scheduled scanning: using pre-configured recurring triggers to scan configurations in the cluster, this is achieved using CronJobs
+* Triggered scanning: Kubescape Operator exposing API to trigger scan of specific objects or the whole cluster
+
+#### Vulnerability scans
+
+Vulnerability scans can be triggered in three ways:
+* Continuous scanning: every new image that are introduced to the cluster is scanned, this is achieved by "watching" Kubernetes API objects
+* Scheduled scanning: using pre-configured recurring triggers to scan for vulnerabilities in the cluster, this is achieved using CronJobs
+* Triggered scanning: Kubescape Operator exposing API to trigger scan of specific image and workloads in the cluster
+
+#### Object storage
+
+In the storage component, full Kubernetes object API is supported with all the CRUD actions.
+
+#### Node agent
+
+*Node agent* is completely event driven and it does not expose a transactional API.
+
+It responds to:
+* Kubernetes API events: Pods that are created, remove on a node
+* Container runtime events: Starting monitoring when a container starts and stopping it when it is stopped.
+* *Node agent* lifecycle events: When the component starts an automatic configuration scan of the node is triggered
+* *OperatorCommand* objects: such objects convey command to the *Node agent* to delete pods/processes as a response to a detection
+* *RuntimeRuleAlertBinding* objects: these object define which detection rules should be applied to which Pods
+
+##### Synchronizer
+
+This component is also completely event driven an its actions are tiggered by two actors:
+* Kubernetes API: *Synchronizer* responds to changes in the Kubernetes API objects - this triggers an action to send these objects to the connected Synchronizer server
+* Synchornizer server: Synchornizer server can trigger the *Synchronizer* component to store objects in Kubernetes API (this is limited by the *ClusterRole* the *Synchronizer* is assigned with)
+
+### Goals
+
+* Kubescape Operator automates scanning and monitoring of Kubernetes clusters
+  * provides up to date security posture data
+  * supplies proposals of how to improve the security posture of the workload and the cluster
+  * provides security observability data
+  * alerts on potential breaches to workloads in the cluster
+* Kubescape CLI scans pre-deployment and post-deployment configurations and images and provides actionable results
+
+#### Security Goals
+
+* All access to the data produced by Kubescape Operator should be only available to those subjects who has the proper RBAC authorizations
+* Kubescape should only share information to Non-Kubernetes endpoints in cases when the user explicitly defined it
+* Kubescape should only have access to objects it must need to complete its job
+* Triggers should only come from authenticated and authorized sources
+
+### Non-goals
+
+* Kubescape is not meant to enforce access control beyond what Kubernetes API already provides
```

Comment:
what about the webhooks in operator?

Reply:
WDYM?

Reply:
we don't provide new RBAC, but we do support validating webhooks in operator